sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -773,13 +836,39 @@ private String buildSQLPlanningErrorMessage(Throwable exception)
       errorMessage = exception.getMessage();
     }
     if (null == errorMessage) {
-      errorMessage = "Please check broker logs for more details";
+      errorMessage = "Please check Broker logs for more details.";
     } else {
       // Re-phrase since planning errors are more like hints
       errorMessage = "Possible error: " + errorMessage;
     }
     // Finally, add the query itself to error message that user will get.
-    return StringUtils.format("Cannot build plan for query: %s. %s", plannerContext.getSql(), errorMessage);
+    return StringUtils.format("Cannot build plan for query. %s", errorMessage);
```

Comment:
having the error message overshadowed by the large query text is indeed annoying. I thought about this before but didn't make any change. One major reason is that many times users are not issuing queries themselves. They are using a BI tool that is building and issuing queries behind the scenes. In such a case, knowing the actual query that was run can be very useful for troubleshooting. 

what do you think? An alternative is to change the order a bit and have `errorMessage` appear before the actual query text. Or we log the whole thing (which we might be doing already)

Reply:
> One major reason is that many times users are not issuing queries themselves. They are using a BI tool that is building and issuing queries behind the scenes. In such a case, knowing the actual query that was run can be very useful for troubleshooting.

That's a good point. Including the query text, but swapping position so it's after the error message, makes sense to me, so the end user gets a copy of the query if it was made through some app. (I think we also do log it, but end users typically don't have access to server logs.)

Reply:
Follow up here: https://github.com/apache/druid/pull/12903